File: Programming-Basics-With-JavaScript/Exams/01-Old-Exams/03-PB-Exam-March-2020/05.careOfPuppy.js
function careOfPuppy(input) {

    const foodQuantityInKg = Number(input[0]);
    let index = 1;
    let current = input[index++];

    const foodQuantityInGrams = foodQuantityInKg * 1000;
    let eatenFoodInGrams = 0;

    while (current !== "Adopted") {
        let currentServing = Number(current);
        eatenFoodInGrams += currentServing;
        current = input[index++];
    }

    const difference = Math.abs(foodQuantityInGrams - eatenFoodInGrams)
    if (foodQuantityInGrams >= eatenFoodInGrams) {
        console.log(`Food is enough! Leftovers: ${difference} grams.`);
    } else {
        console.log(`Food is not enough. You need ${difference} grams more.`);
    }
}
careOfPuppy(["4",
    "130",
    "345",
    "400",
    "180",
    "230",
    "120",
    "Adopted"]);

File: Python_Basics/Exams/07.exam_preparation_jun_2023/05.care_of_puppy.py
food_in_kg = int(input())

food_in_gr = food_in_kg * 1000
is_enough = True
current = input()

while current != 'Adopted':
    current_portion = int(current)
    food_in_gr -= current_portion
    if food_in_gr < 0:
        is_enough = False

    current = input()

if is_enough:
    print(f'Food is enough! Leftovers: {food_in_gr} grams.')
else:
    print(f'Food is not enough. You need {abs(food_in_gr)} grams more.')
